File: EmpresarialesCliente/src/types/Carro.ts
export interface Carro {
  placa: string;                 // id
  marca: string;
  color: string;
  modelo: string;
  anio: number;                  // int
  estado: string;                // NUEVO | USADO | EXCELENTE
  combustible: string;           // GASOLINA | HIBRIDO | ELECTRICO
  numeroPuertas: number;         // int
  tieneAireAcondicionado: boolean;
  precio: number;                // double
  fechaRegistro: string;         // "YYYY-MM-DD HH:mm:ss" (LocalDateTime)
  tipoTransmision: string;       // MANUAL | AUTOMATICA
  tipoVehiculo?: string;         // optional/read-only
  detallesEspecificos?: string;  // optional/read-only
  informacionCompleta?: string;  // optional/read-only
}

export type CarroCreateData = Omit<Carro, 'tipoVehiculo' | 'detallesEspecificos' | 'informacionCompleta'>;

export type CarroUpdateData = CarroCreateData;

export type CarroFilter = {
  placa?: string;
  marca?: string;
  color?: string;
  modelo?: string;
  anio?: number;
  estado?: string;
  combustible?: string;
  tipoTransmision?: string;
  numeroPuertas?: number;
  tieneAireAcondicionado?: boolean;
  precioMin?: number;
  precioMax?: number;
  fechaDesde?: string;
  fechaHasta?: string;
};

// Enums for dropdown options
export const ESTADOS = ['NUEVO', 'USADO', 'EXCELENTE'] as const;
export const COMBUSTIBLES = ['GASOLINA', 'HIBRIDO', 'ELECTRICO'] as const;
export const TRANSMISIONES = ['MANUAL', 'AUTOMATICA'] as const;

File: EmpresarialesClienteCSharp/Models/Carro.cs
using System;
using Newtonsoft.Json;

namespace EmpresarialesClienteCSharp.Models
{
    public class Carro
    {
        [JsonProperty("marca")]
        public string Marca { get; set; } = string.Empty;

        [JsonProperty("color")]
        public string Color { get; set; } = string.Empty;

        [JsonProperty("placa")]
        public string Placa { get; set; } = string.Empty;

        [JsonProperty("combustible")]
        public string Combustible { get; set; } = string.Empty;

        [JsonProperty("modelo")]
        public string Modelo { get; set; } = string.Empty;

        [JsonProperty("anio")]
        public int Anio { get; set; }

        [JsonProperty("estado")]
        public string Estado { get; set; } = string.Empty;

        [JsonProperty("numeroPuertas")]
        public int NumeroPuertas { get; set; }

        [JsonProperty("tieneAireAcondicionado")]
        public bool TieneAireAcondicionado { get; set; }

        [JsonProperty("precio")]
        public double Precio { get; set; }

        [JsonProperty("tipoTransmision")]
        public string TipoTransmision { get; set; } = string.Empty;

        [JsonProperty("fechaRegistro")]
        public DateTime? FechaRegistro { get; set; }

        public override string ToString()
        {
            return $"{Marca} {Modelo} - {Placa}";
        }
    }
}
